hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -422,6 +445,9 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
           case RECORD_INDEX:
             fileGroupCountAndRecordsPair = initializeRecordIndexPartition();
             break;
+          case FUNCTIONAL_INDEX:
+            fileGroupCountAndRecordsPair = initializeFunctionalIndexPartition();
```

Comment:
Does each individual function have its only MDT partition, e.g., `metadata/functional_index_date`, `metadata/functional_index_month`?  Should this consider multiple partitions?

Reply:
Right now, you can jsut create one partition with a create index command (or using async indexer). So, the code assumes that user is not creating multiple functional indexes in the same job but we can extend the functionality in a followup.

Reply:
I see.